### DISCUSSION TITLE
Newbie Question: Scara kinematic choppy movement

### DISCUSSION THREAD

**Discussion by Anno2003** [Paciente8159/uCNC#640] [Q&A] [answered]
when using the scara kinematic, the arm and forearm movements are slow and tend to be not smooth, while the z axis works just fine.  Im currently using 
[this configuration](https://github.com/Paciente8159/uCNC/files/14379080/ucnc_build.json) made using the ucnc config builder on arduino uno with cncshield v3. So far i've tried using the a4988 driver & tmc 2208, while the tmc is considerably smoother it is also moving very slow and somewhat choppy while the a4988 moved reasonably but with very jerky/choppy movement for the arm and forearm. Are there any settings or adjustment that i could use to improve this?

**Comment by Paciente8159** [comment]
Hi there,

This raises several of questions to which I need answers to be able to help you:
- did you used the config files correctly (I get a series of errors with that config, like missing pin definitions, module features not enabled, etc..)?
- this is probably impossible to build on an Arduino UNO (just with the scara kinematics program occupies 99% of the available flash, adding TMC should not be possible)
- are your TMC running on standalone mode or UART mode?
- what are your settings ($$ print out)?

**Reply by Anno2003** [reply]
Sorry apparently i sent the wrong configuration. I meant to use
[this one](https://github.com/Paciente8159/uCNC/files/14381163/ucnc_build.json). I'm trying to use the TMC in standalone mode. The following are my settings output:
```
$0 = 33.333    (Step pulse time, microseconds)
$1 = 0    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 0    (Step direction invert, mask)
$4 = 0    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$7 = 0   
$10 = 1    (Status report options, mask)
$11 = 0.200    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 0    (Hard limits enable, boolean)
$22 = 0    (Homing cycle enable, boolean)
$23 = 0    (Homing direction invert, mask)
$24 = 10.000    (Homing locate feed rate, mm/min)
$25 = 50.000    (Homing search seek rate, mm/min)
$26 = 250    (Homing switch debounce delay, milliseconds)
$27 = 2.000    (Homing switch pull-off distance, millimeters)
$28 = 0.000   
$29 = 0.000   
$30 = 1000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$81 = 0.000   
$100 = 200.000    (X-axis travel resolution, step/mm)
$101 = 200.000    (Y-axis travel resolution, step/mm)
$102 = 200.000    (Z-axis travel resolution, step/mm)
$106 = 100.000   
$107 = 80.000   
$110 = 500.000    (X-axis maximum rate, mm/min)
$111 = 500.000    (Y-axis maximum rate, mm/min)
$112 = 500.000    (Z-axis maximum rate, mm/min)
$120 = 10.000    (X-axis acceleration, mm/sec^2)
$121 = 10.000    (Y-axis acceleration, mm/sec^2)
$122 = 10.000    (Z-axis acceleration, mm/sec^2)
$130 = 200.000    (X-axis maximum travel, millimeters)
$131 = 200.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
ok
```

**Comment by Paciente8159** [comment, marked as answer]
Ok so TMC2208 are being used in standalone mode (same as A4899) and the TMC driver is not needed.

note that A4899 microstepping configuration, is different from TMC2208 (standalone) configuration:

![image](https://github.com/Paciente8159/uCNC/assets/12116536/7700d183-f997-4396-b57e-b3fd4b968868)

And if I'm not mistaken, for the SCARA, the parameters for settings $100 and $101 are actually the total amount of steps a motor needs to perform to do a full rotation. So lets say using a 200 step motor with 1/16 microstepping this results in 3200 steps for a full turn. If you use any kind of gear that ration must also be taken in account.

What machine are you using? A custom machine?

**Reply by Anno2003** [reply]
Yes, i'm using a custom machine with the following design 
<img width="488" alt="image" src="https://github.com/Paciente8159/uCNC/assets/76816549/ec01105c-c46b-40b6-8e81-6fdf15d53801">
it's 40cm tall, and each arm is 30cm. And the stepper is nema17 without additional gear. I'll try setting the $100 and $101 accordingly.

**Reply by Anno2003** [reply]
It works, thankyou for explaining the $100 and $101 settings. I've also managed to get pretty good results by increasing the microstep and increasing the acceleration using the $120 and $121 setting. Im wondering is there any limit for the $120, $121, and $122 settings?

**Reply by Paciente8159** [reply]
There is no limit on the accel settings. The limit will be determined by your machine. With to too high values you may get step skipping, your structure may shake due to high accelerations and deaccelerations, etc... so it depends mainly on your application.
